app/concerns/file_handling.rb

```diff
@@ -0,0 +1,50 @@
+module FileHandling
+  extend ActiveSupport::Concern
+
+  def get_file_pointer(file)
+    { file_pointer: { file: file, agent_id: id } }
+  end
+
+  def get_io(event)
+    return nil unless event.payload['file_pointer'] &&
+                      event.payload['file_pointer']['file'] &&
+                      event.payload['file_pointer']['agent_id']
+    event.user.agents.find(event.payload['file_pointer']['agent_id']).get_io(event.payload['file_pointer']['file'])
```

Comment:
Do you have a spec that asserts that one user's agents cannot access another user's files?  (That fails if this is just `Agent.find`)